src/collection/cereal.rs

```diff
@@ -62,6 +66,28 @@ where
     Ok(map)
 }
 
+impl<'de> Deserialize<'de> for Template {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: Value = Deserialize::deserialize(deserializer)?;
```

Comment:
I'd prefer to keep this format-agnostic, in case we want to support multiple config formats in the future. It's possible using `serde_yaml` would still work with other formats, but it's not best practice. I implemented basically the [same logic in another project using a macro](https://github.com/LucasPickering/env-select/blob/master/src/config/cereal.rs#L22), you could basically copy-paste that. Instead of `ValueSource::from_literal` you can just do `value.to_string()` then parse that as a template, like you are now.

Sorry I didn't link this file when I wrote the issue, if I remembered I had that logic already written somewhere I probably would've just done it myself 😛